### ISSUE TITLE
Higher or Lower resolution? Possible combination of pretrained and custom trained model?

### ISSUE THREAD

**Issue by Aoi-00** [ultralytics/yolov5#8383]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi! I am currently using yolov5 for my final year project, so I am hoping to get some guidance on it. I am using RPI to capture a video stream of a pre-recorded footage being played on a separate screen (ipad/laptop), and streaming it to my PC for yolov5 recognition. Currently, I set the video stream to be **1920x1080** as I thought it would help the yolov5 get better accuracy with higher resolution images, as the captured images are a bit **grainy.**

However, I read that the default pretrained yolov5s model is trained on **640x480 images** so I am wondering if I should have just set the stream to 640x480 for **better accuracy**? Did setting a higher resolution of 1920x1080 actually made the yolov5 accuracy worse? 

Additionally, right now I am only making use of the chair, human, and bottle/bag classes of the cocomo model. If the accuracy does not turn out well for the bottle/bag detection, is there a way to keep the pretrained model for chair & human, while adding a custom trained bottle/bag model and use these 2 on the images? Or would I have to retrain the entire model using a custom dataset for these 4 classes (I'd like to keep the human and chair model as they are working really well now! Thank you for making such an adaptive model!)?

Thank you so much for taking your time to read this and apologies for taking up your time!

**Comment by guillermo-gabrielli-fer** [comment]
The models won't work as well if the resolution is very different from the one they are trained on, if the objects mantain the same  size relative to the image. The recent models that end in 6 (like the YOLOv5m6) are trained on 1280, so they would be more suitable. You can use pseudo-labeling to label the objects on your dataset and then correct them manually.

**Comment by Aoi-00** [comment]
What if I indicate the sizing here as I predict? 
![image](https://user-images.githubusercontent.com/61011887/176429770-8917495f-fbed-4bab-a2b7-187d592d3802.jpeg)

Does it resize the images to 640 for me or is it better that I just set my stream to be 640x480 (are they trained on 640x480?)?

Also, this is the first I've heard of pseudo labelling! How does it work? Does it let me keep the working classes(pretrained human& chair) while editing the bottle & bag classes?

**Comment by glenn-jocher** [comment]
@Aoi-00 the YOLOv5 repo is based on the official YOLOv5 implementation and supports various image sizes, including 640x480. However, it is important to ensure objects maintain the same relative size in the image as the training data for optimal performance.

Regarding pseudo-labeling, it involves using model predictions as "pseudo" ground truth labels, which can be combined with traditional labeled data for further training. This could allow you to fine-tune the model on your specific classes while retaining the pretrained classes.

Feel free to check out our documentation for more information on resizing images and using pseudo-labeling: https://docs.ultralytics.com/yolov5/
